core/os/os.h

```diff
@@ -216,13 +216,19 @@ class OS {
 		uint8_t second;
 	};
 
+	struct DateTime {
+		Date date;
+		Time time;
+	};
```

Comment:
I'm wondering if we should take this a step further, and do this:

```suggestion
	struct DateTime {
		int64_t year;
		Month month;
		uint8_t day;
		Weekday weekday;
		uint8_t hour;
		uint8_t minute;
		uint8_t second;
		bool dst;
	};
```

If so, we wouldn't need `OS.get_date()` and `OS.get_time()`, or the `Date` and `Time` structs. I'm not entirely sure if there are other uses of them that I'm missing though.

Reply:
With this PR, both structs are only used in few `Time::get_X_from_system` methods, so replacing `OS.get_date` and `OS.get_time` is pretty straightforward.
I will update to the PR